fusor-ember-cli/app/controllers/subscriptions/credentials.js

```diff
@@ -53,7 +57,62 @@ export default Ember.Controller.extend({
     } else {
       return 'loginPortal';
     }
-  }.property('model.isAuthenticated')
+  }.property('model.isAuthenticated'),
+
+  isDisconnected: Ember.computed.alias('controllers.deployment.model.is_disconnected'),
+  hasManifestFile: Ember.computed.notEmpty('manifestFile'),
+  noManifestFile: Ember.computed.empty('manifestFile'),
+
+  contentProviderType: function() {
+    return (this.get('isDisconnected') ? "disconnected" : "redhat_cdn");
+  }.property('isDisconnected'),
+
+  contentProviderTitle: function() {
+    return (this.get('isDisconnected') ? "Disconnected" : "Red Hat CDN");
+  }.property('isDisconnected'),
+
+  isDisconnectedSelected: function() {
+    return (this.get('contentProviderType') === 'disconnected');
+  }.property('contentProviderType'),
+
+  actions: {
+    providerTypeChanged: function() {
+      return this.set('isDisconnected', this.get('isDisconnectedSelected'));
+    },
+
+    uploadManifest: function() {
+      var self = this;
+      var manifestFile = document.getElementById('manifest-file-field').files[0];
+      var formData = new FormData();
+      formData.append('manifest_file[name]', manifestFile.name);
+      formData.append('manifest_file[file]', manifestFile);
+      formData.append('manifest_file[deployment_id]', this.get('deploymentId'));
```

Comment:
content-type is 'multipart/form-data', not json
I'm passing `deployment_id` as what of the payload